src/EditorFeatures/Test/Completion/FileSystemCompletionHelperTests.cs

```diff
@@ -0,0 +1,234 @@
+                });
+
+            // Note backslashes in description are escaped
+            AssertItemsEqual(fsc.GetItems(@"", CancellationToken.None),
+                @"'X:', Folder, 'Text|X:'",
+                @"'\\', , 'Text|\5C\5C'",
+                @"'1', Folder, 'Text|X:\5CA\5C1'",
+                @"'2', Folder, 'Text|X:\5CA\5C2'",
+                @"'3', Folder, 'Text|X:\5CA\5C3'",
+                @"'file5.abc', File, C#, 'Text|X:\5CB\5Cfile5.abc'");
+        }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void GetItems_Windows_NoSearchPaths()
+        {
+            var fsc = new TestFileSystemCompletionHelper(
+                searchPaths: ImmutableArray<string>.Empty,
+                baseDirectoryOpt: null,
+                allowableExtensions: ImmutableArray.Create(".abc", ".def"),
+                drives: new[] { @"X:\" },
+                directories: new[]
+                {
+                    @"X:",
+                    @"X:\A",
+                    @"X:\A\1",
+                    @"X:\A\2",
+                    @"X:\A\3",
+                    @"X:\B",
+                },
+                files: new[]
+                {
+                    @"X:\A\1\file1.abc",
+                    @"X:\A\2\file2.abc",
+                    @"X:\B\file4.x",
+                    @"X:\B\file5.abc",
+                    @"X:\B\hidden.def",
+                });
+
+            // Note backslashes in description are escaped
+            AssertItemsEqual(fsc.GetItems(@"", CancellationToken.None),
+                @"'X:', Folder, 'Text|X:'",
+                @"'\\', , 'Text|\5C\5C'");
+        }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void GetItems_Windows_Network()
+        {
+            var fsc = new TestFileSystemCompletionHelper(
+                searchPaths: ImmutableArray<string>.Empty,
+                baseDirectoryOpt: null,
+                allowableExtensions: ImmutableArray.Create(".cs"),
+                drives: Array.Empty<string>(),
+                directories: new[]
+                {
+                    @"\\server\share",
+                    @"\\server\share\C",
+                    @"\\server\share\D",
+                },
+                files: new[]
+                {
+                    @"\\server\share\C\b.cs",
+                    @"\\server\share\C\c.cs",
+                    @"\\server\share\D\e.cs",
+                });
+
+            AssertItemsEqual(fsc.GetItems(@"\\server\share\", CancellationToken.None),
+                @"'C', Folder, 'Text|\5C\5Cserver\5Cshare\5CC'",
+                @"'D', Folder, 'Text|\5C\5Cserver\5Cshare\5CD'");
+
+            AssertItemsEqual(fsc.GetItems(@"\\server\share\C\", CancellationToken.None),
+                @"'b.cs', File, C#, 'Text|\5C\5Cserver\5Cshare\5CC\5Cb.cs'",
+                @"'c.cs', File, C#, 'Text|\5C\5Cserver\5Cshare\5CC\5Cc.cs'");
+        }
+
+        [ConditionalFact(typeof(UnixLikeOnly))]
```

Comment:
Why do we have a unix test in the editor layer? We don't run these tests on Unix.